libc/test/UnitTest/LibcTestMain.cpp

```diff
@@ -43,7 +43,15 @@ TestOptions parseOptions(int argc, char **argv) {
 
 } // anonymous namespace
 
-extern "C" int main(int argc, char **argv, char **envp) {
+// The C++ standard forbids declaring the main function with a linkage specifier
+// outisde of 'freestanding' mode, only define the linkage for hermetic tests.
+#ifdef LIBC_HERMETIC_TEST
```

Comment:
Shall we detect `-ffreestanding` with `__STD_HOSTED__ == 0` instead?

Reply:
That's fair, good point.